src/srcShared/fs.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+
+const fs = vscode.workspace.fs
+type Uri = vscode.Uri
+
+/**
+ * @warning Do not import this class specifically, instead import the instance {@link fsCommon}.
+ *
+ * This class contains file system methods that are "common", meaning
+ * it can be used in the browser or desktop.
+ *
+ * Technical Details:
+ * TODO: Verify the point below once I get this hooked up to the browser code.
+ * - vscode.workspace.fs dynamically resolves the correct file system provider
+ *   to use. This means that in the browser it will attempt to use the File System
+ *   Access API.
+ *
+ *
+ * MODIFYING THIS CLASS:
+ * - All methods must work for both browser and desktop
+ * - Do not use 'fs' or 'fs-extra' since they are not browser compatible.
+ *   If they have functionality that cannot be achieved with a
+ *   browser+desktop implementation, then create it in the module {@link TODO}
+ */
+export class FileSystemCommon {
+    constructor() {}
+
+    async mkdir(path: Uri | string): Promise<void> {
+        path = FileSystemCommon.getUri(path)
+        return fs.createDirectory(path)
+    }
+
+    async readFile(path: Uri | string): Promise<Uint8Array> {
+        path = FileSystemCommon.getUri(path)
+        return fs.readFile(path)
+    }
+
+    async readFileAsString(path: Uri | string): Promise<string> {
+        path = FileSystemCommon.getUri(path)
+        return FileSystemCommon.arrayToString(await this.readFile(path))
+    }
+
+    async appendFile(path: Uri | string, content: Uint8Array | string): Promise<void> {
+        path = FileSystemCommon.getUri(path)
+
+        const currentContent: Uint8Array = (await this.fileExists(path)) ? await this.readFile(path) : new Uint8Array(0)
+        const currentLength = currentContent.length
+
+        const newContent = FileSystemCommon.asArray(content)
+        const newLength = newContent.length
+
+        const finalContent = new Uint8Array(currentLength + newLength)
+        finalContent.set(currentContent)
+        finalContent.set(newContent, currentLength)
+
+        return this.writeFile(path, finalContent)
+    }
+
+    async fileExists(path: Uri | string): Promise<boolean> {
+        path = FileSystemCommon.getUri(path)
+        const stat = await this.stat(path)
+        return stat === undefined ? false : stat.type === vscode.FileType.File
+    }
+
+    async writeFile(path: Uri | string, data: string | Uint8Array): Promise<void> {
+        path = FileSystemCommon.getUri(path)
+        return fs.writeFile(path, FileSystemCommon.asArray(data))
+    }
+
+    /**
+     * @returns the stat of the file or undefined if it does not exist
```

Comment:
this docstring is not telling us anything the signature didn't already tell us :)

Reply:
I figured this was different since it may not be clear `undefined` is only for the case where it does not exist. If there was an error, eg permissions, we would throw